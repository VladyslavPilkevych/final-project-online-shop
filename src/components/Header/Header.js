import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { toggleMenu } from '../../store/actionCreators/menuAC';
import { toggleSearch } from '../../store/actionCreators/searchAC';
import { toggleCart } from '../../store/actionCreators/cartAC';
import { toggleMiniMenu } from '../../store/actionCreators/miniMenuAC';

import Menu from '../Menu/Menu';
import MiniMenu from '../MiniMenu/MiniMenu';
import Search from '../Search/Search';
import MiniCart from '../MiniCart/MiniCart';
import Avatar from '../Avatar/Avatar';
import { ReactComponent as FbIcon } from '../../assets/icons/ant-design_facebook-filled.svg';
import { ReactComponent as InstagramIcon } from '../../assets/icons/ant-design_instagram-filled.svg';
import { ReactComponent as LogoIcon } from '../../assets/icons/Logo.svg';
import { ReactComponent as CartIcon } from '../../assets/icons/Cart.svg';
import { ReactComponent as SearchIcon } from '../../assets/icons/Vector.svg';

import styles from './Header.module.scss';

const items = [
  {
    value: 'Laptops',
    to: '/filter/laptop',
    className: '{styles.navLinks}',
  },
  {
    value: 'Monitors',
    to: '/filter/monitor',
    className: '{styles.navLinks}',
  },
  {
    value: 'Phones',
    to: '/filter/phones',
    className: '{styles.navLinks}',
  },
  {
    value: 'Headphones',
    to: '/filter/headphones',
    className: '{styles.navLinks}',
  },
  // {
  //   value: 'Filter',
  //   to: '/filter',
  //   className: '{styles.navLinks}',
  // },
];

function Header() {
  const isOpen = useSelector((state) => state.menu.isOpen);
  const isOpenMiniMenu = useSelector((state) => state.miniMenu.isOpenMiniMenu);

  const isOpenSearch = useSelector((state) => state.search.isOpenSearch);
  const isOpenCart = useSelector((state) => state.cart.isOpenCart);
  const [value, setValue] = useState('');

  const dispatch = useDispatch();

  return (
    <header>
      <div className={styles.headerWrapper}>
        <div className={styles.blackBg}>
          <div>
            <p className={styles.dateBur}>
              Mon-Thu:
              <span className={styles.timeBur}> 9:00 AM - 5:30 PM</span>
            </p>
          </div>
          <div className={styles.adressBurWrapper}>
            <p className={styles.adressBur}>Visit our showroom in 1234 Street Adress City A</p>
            <a className={styles.adressBurLink} href="mailto:address@gmail.com">
              Contact Us
            </a>
          </div>
          <div className={styles.phoneBurWrapper}>
            <div>
              <a href="tel:0012345678" className={styles.phoneNumber}>
                Call Us: (00) 1234 5678
              </a>
            </div>
            <div>
              <FbIcon className={styles.fbIcon} />
              <InstagramIcon className={styles.instagramIcon} />
            </div>
          </div>
        </div>
      </div>
      <div className={styles.whiteWrapper}>
        <div className={styles.wrapperContainer}>
          <div className={styles.wrapperNavBar}>
            <nav className={styles.navBur}>
              <NavLink className={styles.navLinksLogo} to="/">
                <LogoIcon className={styles.logo} role="button" tabIndex="0" />
              </NavLink>
            </nav>
            <nav className={styles.navBur}>
              {!isOpenMiniMenu && (
                <div className={styles.btnBurger} role="button" tabIndex="0" onClick={() => dispatch(toggleMiniMenu(!isOpenMiniMenu))}>
                  {' '}
                  <span className={styles.btnSpan}> </span>
                </div>
              )}
              <Menu items={items} />
              <MiniMenu items={items} />
            </nav>
            {/* <nav className={styles.navBur}>
              {items.map((item) => (
                <li key={item.value}>
                  <div>
                    <NavLink className={styles.navLinks} style={items.style} to={item.to}>
                      {item.value}
                    </NavLink>
                  </div>
                </li>
              ))}
            </nav> */}
          </div>
          <Search />
          <div className={styles.searchBox}>
            <input
              className={styles.searchInput}
              type="text"
              placeholder="Search entiere store here..."
              value={value}
              onChange={(e) => {
                setValue(e.target.value);
                console.log(value);
              }}
            />
          </div>
          <div className={styles.navBarRight}>
            <nav className={styles.navBur}>
              <SearchIcon
                className={styles.searchIcon}
                role="button"
                tabIndex="0"
                onClick={() => {
                  dispatch(toggleSearch(!isOpenSearch));
                  dispatch(toggleMenu(!isOpen));
                }}
              />
              <li className={styles.navBarRightItem}>
                <CartIcon className={styles.cartIcon} role="button" tabIndex="0" onClick={() => dispatch(toggleCart(!isOpenCart))} />
                <MiniCart />
              </li>
              <li className={styles.navBarRightItem}>
                <Avatar />
              </li>
            </nav>
          </div>
        </div>
      </div>
    </header>
  );
}

export default Header;