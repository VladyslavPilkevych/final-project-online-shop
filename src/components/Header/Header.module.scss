@import '../../assets/styles/global.module.scss';
header {
  // position: fixed;
  // z-index: 20;
  top: 0;
  left: 0;
  width: 100%;
}
.headerWrapper {
  background-color: $black;
}
.blackBg {
  display: flex;
  margin: 0 auto;
  max-width: 1398px;
  padding-left: 10px;
  padding-right: 10px;
  align-items: center;
  justify-content: space-between;
}
.dateBur {
  font-size: 12px;
  font-weight: 600;
  line-height: 18px;
  color: $dark-grey-text;
}
.timeBur {
  color: $white;
  font-size: 12px;
  font-weight: 600;
  line-height: 18px;
}
.adressBurWrapper {
  display: flex;
}
.adressBur {
  font-size: 12px;
  font-weight: 600;
  color: $grey-text;
  line-height: 18px;
}
.adressBurLink {
  color: $white;
  font-size: 12px;
  font-weight: 600;
  line-height: 18px;
  padding-left: 4px;
}
.phoneNumber {
  color: $white;
  font-size: 12px;
  margin-top: 15px;
  font-weight: 600;
  line-height: 18px;
  margin-right: 14px;
  margin-bottom: 15px;
  display: inline-block;
  text-decoration: none;
}
.phoneBurWrapper {
  display: flex;
}
.fbIcon {
  margin-top: 13px;
  display: inline-block;
}
.instagramIcon {
  margin-top: 13px;
  margin-left: 8px;
  display: inline-block;
}
.whiteWrapper {
  background-color: $white;
  border-bottom: 1px solid $grey-star;
}
.wrapperContainer {
  display: flex;
  margin: 0 auto;
  max-width: 1398px;
  padding-left: 10px;
  padding-right: 10px;
  align-items: center;
  justify-content: space-between;
}
.wrapperNavBar {
  display: flex;
}
.navBur {
  display: flex;
  margin: 0 auto;
  flex-basis: 5%;
  list-style: none;
  align-items: center;
}
nav {
  li {
    a {
      cursor: pointer;
      text-decoration: none;
    }
  }
}
.navLinks {
  color: $black;
  font-size: 14px;
  font-weight: 600;
  line-height: 21px;
  margin-right: 30px;
}
.navLinks:hover {
  color: $blue;
}
.logo {
  fill: $blue;
  outline: none;
  cursor: pointer;
  margin-top: 13px;
  margin-bottom: 4px;
}
.navBarRight {
  display: flex;
  position: relative;
}
.navBarRightItem {
  margin-left: 25px;
}
.cartIcon {
  fill: $black;
  outline: none;
  cursor: pointer;
  outline: none;
}
.signInIcon {
  stroke: $black;
}
.searchIcon {
  fill: $black;
  outline: none;
  cursor: pointer;
}
.searchBox {
  display: none;
}
.navLinks {
  color: $black;
  font-size: 14px;
  font-weight: 600;
  line-height: 21px;
  margin-right: 30px;
  text-decoration: none;
}
.cartIconIndex {
  width: 16px;
  height: 16px;
  background-color: $blue;
  border-radius: 50%;
  color: $white;
  text-align: center;
  font-weight: 700;
  font-size: 10px;
  line-height: 17px;
  position: absolute;
  bottom: 24px;
  right: 46px;
}

@media (max-width: 768px) and (min-width: 376px) {
  .blackBg {
    max-width: 768px;
  }
  .dateBur {
    font-size: 11px;
    line-height: 16px;
    padding-top: 10px;
    margin-left: 100px;
    padding-bottom: 10px;
  }
  .timeBur {
    font-size: 11px;
    line-height: 16px;
  }
  .phoneBurWrapper {
    display: none;
  }
  .whiteWrapper {
    background-color: $blue;
  }
  .logo {
    top: -6px;
    left: 19px;
    width: 46px;
    height: 41px;
    fill: $white;
    position: absolute;
    background-color: $blue;
    clip-path: ellipse(50% 59% at 50% 64%);
    cursor: pointer;
  }
  .adressBur {
    display: none;
  }
  .adressBurLink {
    font-size: 11px;
    line-height: 16px;
    margin-right: 28px;
  }
  .wrapperContainer {
    align-items: center;
  }
  .navBarRight {
    padding: 16px 0;
    margin-right: 23px;
  }
  .cartIcon {
    fill: $white;
  }
  .signInIcon {
    stroke: $white;
  }
  .searchBox {
    display: flex;
    flex-basis: 70%;
    margin-left: 30px;
    flex-direction: column;
  }
  .searchInput {
    width: 100%;
    height: 38px;
    border: none;
    outline: none;
    padding: 6px;
    font-size: 11px;
    font-weight: 400;
    line-height: 16px;
    border-radius: 25px;
    transition: 0.5s ease;
    color: $dark-grey-text;
    box-sizing: border-box;
    background-color: white;
  }
  .searchIcon {
    display: none;
  }
  .btnBurger {
    z-index: 22;
    cursor: pointer;
    margin-left: 10px;
    position: relative;
  }
  .btnBurger::after {
    top: -6px;
    z-index: 22;
    width: 17px;
    content: ' ';
    color: $white;
    position: absolute;
    border-bottom: 2px solid $white;
    height: 2px;
  }
  .btnBurger::before {
    top: 6px;
    width: 17px;
    height: 2px;
    z-index: 22;
    content: ' ';
    color: $white;
    position: absolute;
    border-bottom: 2px solid $white;
  }
  .btnSpan {
    width: 17px;
    height: 2px;
    color: $white;
    position: absolute;
    border-bottom: 2px solid $white;
  }
  .navLinks {
    display: none;
  }
  .cartIconIndex {
    right: 46px;
    bottom: 38px;
    color:  $blue;
    border-radius: 50%;
    position: absolute;
    background-color: $white;
  }
}
@media (max-width: 375px) and (min-width: 240px) {
  .navLinks {
    display: none;
  }

  .blackBg {
    max-width: 375px;
  }
  .dateBur {
    font-size: 11px;
    line-height: 16px;
    padding-top: 10px;
    margin-left: 70px;
    padding-bottom: 10px;
  }
  .phoneBurWrapper {
    display: none;
  }
  .whiteWrapper {
    background-color: $blue;
  }
  .logo {
    top: -6px;
    left: 19px;
    width: 46px;
    height: 41px;
    fill: $white;
    cursor: pointer;
    position: absolute;
    background-color: $blue;
    clip-path: ellipse(50% 59% at 50% 64%);
  }
  .adressBur {
    display: none;
  }
  .adressBurLink {
    font-size: 11px;
    line-height: 16px;
    margin-right: 18px;
  }
  .navBarRight {
    padding: 16px 0;
    margin-right: 23px;
  }
  .cartIcon {
    fill: $white;
  }
  .signInIcon {
    stroke: $white;
  }
  .searchBox {
    display: flex;
    flex-basis: 70%;
    margin-left: 30px;
  }
  .searchIcon {
    display: none;
  }
  .searchInput {
    width: 100%;
    height: 38px;
    border: none;
    outline: none;
    padding: 6px;
    font-size: 11px;
    font-weight: 400;
    line-height: 16px;
    border-radius: 25px;
    transition: 0.5s ease;
    color: $dark-grey-text;
    box-sizing: border-box;
    background-color: white;
  }
  .btnBurger {
    z-index: 22;
    cursor: pointer;
    position: relative;
  }
  .btnBurger::after {
    top: -6px;
    z-index: 22;
    width: 17px;
    height: 2px;
    content: ' ';
    color: $white;
    position: absolute;
    border-bottom: 2px solid $white;
  }
  .btnBurger::before {
    top: 6px;
    width: 17px;
    height: 2px;
    z-index: 22;
    content: ' ';
    color: $white;
    position: absolute;
    border-bottom: 2px solid $white;
  }
  .btnSpan {
    width: 17px;
    height: 2px;
    color: $white;
    position: absolute;
    border-bottom: 2px solid $white;
  }
  .cartIconIndex {
    right: 46px;
    bottom: 38px;
    color:  $blue;
    border-radius: 50%;
    position: absolute;
    background-color: $white;
  }
}
