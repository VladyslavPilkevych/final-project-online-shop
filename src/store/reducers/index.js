import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
