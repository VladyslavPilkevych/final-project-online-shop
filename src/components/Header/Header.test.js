/* eslint-disable react/react-in-jsx-scope */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';
import Header from './Header';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <Header />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot header should render', () => {
  test('Snapshot header is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
