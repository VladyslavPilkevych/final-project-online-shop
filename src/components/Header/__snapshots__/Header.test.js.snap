// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot header should render Snapshot header is rendered 1`] = `
<DocumentFragment>
  <header>
    <div
      class="headerWrapper"
    >
      <div
        class="blackBg"
      >
        <div>
          <p
            class="dateBur"
          >
            Mon-Thu:
            <span
              class="timeBur"
            >
               9:00 AM - 5:30 PM
            </span>
          </p>
        </div>
        <div
          class="adressBurWrapper"
        >
          <p
            class="adressBur"
          >
            Visit our showroom in 1234 Street Adress City A
          </p>
          <a
            class="adressBurLink"
            href="mailto:address@gmail.com"
          >
            Contact Us
          </a>
        </div>
        <div
          class="phoneBurWrapper"
        >
          <div>
            <a
              class="phoneNumber"
              href="tel:0012345678"
            >
              Call Us: (00) 1234 5678
            </a>
          </div>
          <div>
            <svg
              class="fbIcon"
            >
              ant-design_facebook-filled.svg
            </svg>
            <svg
              class="instagramIcon"
            >
              ant-design_instagram-filled.svg
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="whiteWrapper"
    >
      <div
        class="wrapperContainer"
      >
        <div
          class="wrapperNavBar"
        >
          <nav
            class="navBur"
          >
            <a
              aria-current="page"
              class="navLinksLogo active"
              href="/"
            >
              <svg
                class="logo"
                role="button"
                tabindex="0"
              >
                Logo.svg
              </svg>
            </a>
          </nav>
          <nav
            class="navBur"
          >
            <div
              class="btnBurger"
              role="button"
              tabindex="0"
            >
               
              <span
                class="btnSpan"
              >
                 
              </span>
            </div>
            <div
              class="menu"
              role="button"
              tabindex="0"
            >
              <div
                class="menuWrapper"
              >
                <div
                  class="menuContent"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="menuHeader"
                  >
                    <svg
                      class="menuLogoIcon"
                    >
                      Logo.svg
                    </svg>
                    <svg
                      class="btnClose"
                    >
                      closeIcon.svg
                    </svg>
                  </div>
                  <ul
                    class="navBur"
                  >
                    <li>
                      <div>
                        <a
                          class="navLinks"
                          href="/filter/laptop"
                        >
                          Laptops
                        </a>
                      </div>
                    </li>
                    <li>
                      <div>
                        <a
                          class="navLinks"
                          href="/filter/monitor"
                        >
                          Monitors
                        </a>
                      </div>
                    </li>
                    <li>
                      <div>
                        <a
                          class="navLinks"
                          href="/filter/phones"
                        >
                          Phones
                        </a>
                      </div>
                    </li>
                    <li>
                      <div>
                        <a
                          class="navLinks"
                          href="/filter/headphones"
                        >
                          Headphones
                        </a>
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </nav>
        </div>
        <div
          class="searchBox"
        >
          <form>
            <input
              class="searchInput"
              placeholder="Search entiere store here..."
              type="text"
              value=""
            />
          </form>
        </div>
        <div
          class="navBarRight"
        >
          <nav
            class="navBur"
          >
            <svg
              class="searchIcon"
              role="button"
              tabindex="0"
            >
              Vector.svg
            </svg>
            <li
              class="navBarRightItem"
            >
              <svg
                class="cartIcon"
                role="button"
                tabindex="0"
              >
                Cart.svg
              </svg>
            </li>
            <li
              class="navBarRightItem"
            >
              <div>
                <a
                  class=""
                  href="/sign-in"
                >
                  <svg
                    class="signInIcon"
                  >
                    Sign-in.svg
                  </svg>
                </a>
              </div>
            </li>
          </nav>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
